src/pybind/mgr/dashboard/frontend/src/app/core/auth/user-form/user-form.component.spec.ts

```diff
@@ -101,6 +101,41 @@ describe('UserFormComponent', () => {
       formHelper.expectValidChange('confirmpassword', 'aaa');
     });
 
+    it('should validate password strength very strong', () => {
+      formHelper.setValue('password', 'testpassword#!$!@$');
+      component.checkPassword();
+      expect(component.passwordStrengthDescription).toBe('Very strong');
+      expect(component.passwordStrengthLevel).toBe('passwordStrengthLevel4');
+    });
+
+    it('should validate password strength strong', () => {
+      formHelper.setValue('password', 'testpassword0047!@');
+      component.checkPassword();
+      expect(component.passwordStrengthDescription).toBe('Strong');
+      expect(component.passwordStrengthLevel).toBe('passwordStrengthLevel3');
+    });
+
+    it('should validate password strength ok ', () => {
+      formHelper.setValue('password', 'mypassword1!@');
+      component.checkPassword();
+      expect(component.passwordStrengthDescription).toBe('OK');
+      expect(component.passwordStrengthLevel).toBe('passwordStrengthLevel2');
+    });
+
+    it('should validate password strength weak', () => {
+      formHelper.setValue('password', 'mypassword1');
```

Comment:
Any update on this?

Reply:
I've added comment in general in this PR that I updated the requierements. 
Yes it can be allowed, because it was changed for the requirement that the password need to have 10 cerdits, it can contain 10 letters, numbers etc, but all aren't required.

Reply:
The question I'm referring to is the following:

Shouldn't the dashboard tell the user that he can't use such a password for some reason instead of rating his password complexity?
It would be a better user experience, if the user knows whats missing for his password to be valid.


Now a short story about my experience with password ratings a view days ago:
I encountered such a problem a view days ago, where my password, generated by a password generator with 16 chars, was rated "not so good", but I wanted to proceed anyway, after clicking submit I got the message that I also have to add non alphanumeric characters, which I disabled in my password generator as many sites can't handle *any* other character equally well.
Than I thought - OK - 24 chars with any symbol, rated "pretty good" - hit submit, got the error that was saying the password was too long...
I didn't register because at this point I had enough of it. If they had just told me what they want before hitting submit I would have registered.

Reply:
The user has information what are the requirements for the password in a short message as below: 
Required  rules for password complexity:
  - cannot contain username
  - cannot contain any keyword used in Ceph
  - must  consist of characters from the following groups:
    * alphabetic a-z, A-Z
    * numbers 0-9
    * special chars: !"#$%& \'()*+,-./:;<=>?@[\\]^_`{{|}}~
    * any other characters (signs)

There was changed that the password don't need to contain at least one small letter, capital letter, number, special character, but it is said that it has to be strong, therefore we can't tell the user what exactly he should add. If he the password is at least weak then it will be changed to that one.

Reply:
Hi, just checked out the current version.
The help sign is missing "Cannot contain repetitive characters", as I got this  error message on the first try. On the other hand I was able to create a password with only 2 characters which were the same (space).

I'm really not sure if a password with 2 characters is as strong as a password...
Compared with the equal of 10 characters that contain only characters.

https://password.kaspersky.com/ is a good site to check password strength (tested a view)

Reply:
I will provide a patch soon.

Reply:
I can add minimum length with 8 characters, except that rules and I think it will be good. Because with mypassword1, the problem is that it is commonly used. Maybe it is better to add to the "forbiden words" list words like password  etc.

Reply:
If you could add some minimum length that would be fine :)

I know we could use some dictionaries to test against, but it's sufficient enough not to allow the same host to brute force a password by restricting the login attempts (which we do, if remember correctly)

Reply:
ok, therefore I will add the minimum length.